### DISCUSSION TITLE
react-native-audio on ios 17 record aac failed, the file is only 33bytes size

### DISCUSSION THREAD

**Discussion by mikedream89** [community/community#128921] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

react-native-audio on ios 17 record aac failed, the file is only 33bytes size

**Comment by ArchangelSavi** [comment]
Upon attempting to record audio in AAC format using the react-native-audio library on iOS 17, encountered issue persists where the resulting file size is only 33 bytes, devoid of valid audio data. This anomaly is observed exclusively on devices running iOS 17.

Steps to Reproduce:
Initiate a React Native project integrating react-native-audio.
Implement audio recording functionality leveraging AAC format on an iOS 17 device.
Observe the recorded file, confirming it is limited to 33 bytes without containing any discernible audio content.
Potential Causes:
iOS 17 Updates: Recent updates in iOS 17 may have introduced changes impacting the recording mechanisms or the interoperability of react-native-audio.

Library Compatibility: Potential lack of full compatibility between the react-native-audio library and iOS 17's audio recording APIs or file format handling.

Permission Dynamics: Revised permission requirements in iOS 17 affecting the library's access to necessary resources, such as microphone usage permissions.

Troubleshooting Steps:
Verify Permissions:
Confirm that the application has been granted appropriate permissions to access the device's microphone. This includes updating the Info.plist file with the necessary microphone usage description string.

xml
Copy code
<key>NSMicrophoneUsageDescription</key>
<string>We need access to the microphone to record audio.</string>
Update Library:
Ensure utilization of the latest version of react-native-audio. Refer to the library's official repository for any recent updates or patches addressing compatibility with iOS 17.

Consider Alternative Solutions:
In cases where the issue persists despite adherence to permissions and library updates, explore alternative audio recording libraries compatible with iOS 17 or consider consulting community forums and developer resources for potential solutions.

By systematically addressing these aspects, including verifying permissions, updating the library, and exploring alternative solutions, resolution of the AAC recording issue on iOS 17 with react-native-audio can be pursued effectively.

**Comment by SaviPandey** [comment]
When attempting to record audio in AAC format using react-native-audio on iOS 17, the resulting file is only 33 bytes in size and lacks valid audio data.

Steps to Reproduce:
Set up a React Native project with react-native-audio.
Implement AAC audio recording on an iOS 17 device.
Verify the recorded file, noting its inadequate size and absence of audio content.
Potential Causes:
iOS 17 Updates: Recent iOS 17 updates may have impacted react-native-audio's compatibility with AAC recording.

Library Compatibility: Possible issues with how react-native-audio interacts with iOS 17's AAC recording APIs.

Permission Dynamics: Updated permissions in iOS 17 affecting microphone access for audio recording.

Troubleshooting Steps:
Verify Permissions: Ensure the app has correct microphone permissions by updating the Info.plist file with appropriate usage descriptions.

xml
Copy code
<key>NSMicrophoneUsageDescription</key>
<string>We need access to the microphone to record audio.</string>
Update Library: Use the latest version of react-native-audio and check for updates addressing iOS 17 compatibility on the library's repository.

Consider Alternatives: If issues persist, explore alternative libraries or consult community forums for solutions.

By addressing these aspects methodically, resolving the AAC recording issue on iOS 17 with react-native-audio can be pursued effectively.

**Comment by ebndev** [comment]
Hi @mikedream89, thanks for being a part of the GitHub Community!

The Profile category is for conversations related to your GitHub profile page. I have moved this post to General, however, it looks like you're looking for support from the [react-native-audio](https://github.com/jsierles/react-native-audio) repo, but I do see that it is archived.

We want to make sure you’re getting the best support you can and that your message goes where you intended, but this may not be the right place.

Best of luck!
